content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -682,6 +682,95 @@ The new Job itself will have a different uid from `a8f3d00d-c6d2-11e5-9f87-42010
 `manualSelector: true` tells the system that you know what you are doing and to allow this
 mismatch.
 
+### Custom pod failure policy {#pod-failure-policy}
+
+{{< feature-state for_k8s_version="v1.25" state="alpha" >}}
+
+{{< note >}}
+You can only configure a Pod failure policy for a Job if you have the
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+enabled in your cluster: `JobPodFailurePolicy`. Additionally, it is recommended
+to enable the `PodDisruptionsCondition` feature gate in order to be able to detect and handle
+pod disruption conditions in the Pod failure policy (see also:
+[Pod disruption conditions](/docs/concepts/workloads/pods/disruptions#pod-disruption-conditions)). Both feature gates are
+available in Kubernetes v1.25.
+{{< /note >}}
+
+In some situations you may want to have a better control over handling of Pod
+failures than the default policy based on the Job's `.spec.backoffLimit` gives
+(see: [Pod backoff failure policy](#pod-backoff-failure-policy)).
+
+For example, you may want to terminate a job as soon as one of its Pods fails
+with an exit code indicating a software bug. This may allow you to cut down costs of running computational
+workloads by avoiding unnecessary Pod restarts. On the other hand, you may want
+not to count pod failures caused by disruptions (such as preemption, eviction API
+or taint-based Pod deletions) towards the `.spec.backoffLimit` counter.
+
+You can satisfy the above use cases by specifying a Pod failure policy
+(in the `.spec.podFailurePolicy` field) for handling Pod failures based on the
+container exit codes and the Pod conditions.
+
+Here is a manifest for a Job that defines a `podFailurePolicy`:
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+spec:
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: example
+        image: example-job-image:v1
+        ...
+  backoffLimit: 6
+  podFailurePolicy:
+    rules:
+    - action: FailJob
+      onExitCodes:
+        containerName: example   # optional
+        operator: In             # one of: In, NotIn
+        values: [42]
+    - action: Ignore             # one of: Ignore, FailJob, Count
+      onPodConditions:
+      - type: DisruptionTarget   # indicates Pod disruption
+```
+
+The example manifest exclude Pod disruptions from counting towards the
+`.spec.backoffLimit` counter. For the `example` container specifically:
+
+- an exit status of 0 means that the container succeeded
+- an exit status of 42 means that the **entire Job** has failed
+- any other exit status represents a fault that is specific to the `example` container.
```

Comment:
maybe explain what would happen if the `example` container exits with a value different than 42 or 0. And what happens with other containers.

Reply:
Rephrased this bit